agent/src/beerocks/slave/ap_manager_thread.cpp

```diff
@@ -441,6 +441,70 @@ bool ap_manager_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_
     }
 
     switch (beerocks_header->action_op()) {
+    case beerocks_message::ACTION_APMANAGER_ENABLE_APS_REQUEST: {
+        LOG(TRACE) << "ACTION_APMANAGER_ENABLE_APS_REQUEST";
+
+        auto notification =
+            cmdu_rx.addClass<beerocks_message::cACTION_APMANAGER_ENABLE_APS_REQUEST>();
+        if (!notification) {
+            LOG(ERROR) << "addClass cACTION_APMANAGER_ENABLE_APS_REQUEST failed";
+            return false;
+        }
+
+        auto response =
+            message_com::create_vs_message<beerocks_message::cACTION_APMANAGER_ENABLE_APS_RESPONSE>(
+                cmdu_tx);
+
+        if (!response) {
+            LOG(ERROR) << "Failed building message!";
+            return false;
+        }
+
+        response->success() = true;
+
+        // Enable beaconing by setting start_disabled flag to false.
+        if (!ap_wlan_hal->set_start_disabled(false)) {
+            LOG(ERROR) << "Failed setting start_disabled";
+            response->success() = false;
+            message_com::send_cmdu(slave_socket, cmdu_tx);
+            break;
+        }
+
+        // Disable the radio interface to make hostapd to consider the new configuration.
+        if (!ap_wlan_hal->disable()) {
+            LOG(ERROR) << "Failed disable";
+            response->success() = false;
+            message_com::send_cmdu(slave_socket, cmdu_tx);
+            break;
+        }
+
+        // If it is not the radio of the BH, then channel, bandwidth and center channel paramenters
+        // will be all set to 0.
+        LOG(DEBUG) << "Setting AP channel: "
+                   << ", channel=" << int(notification->channel())
+                   << ", bandwidth=" << int(notification->bandwidth())
+                   << ", center_channel=" << int(notification->center_channel());
+
+        // Set original channel or BH channel
+        if (!ap_wlan_hal->set_channel(notification->channel(), notification->bandwidth(),
+                                      notification->center_channel())) {
+            LOG(ERROR) << "Failed setting set_channel";
+            response->success() = false;
+            message_com::send_cmdu(slave_socket, cmdu_tx);
+            break;
+        }
+
+        // Enable the radio interface to apply the new configuration.
+        if (!ap_wlan_hal->enable()) {
+            LOG(ERROR) << "Failed enable";
+            response->success() = false;
+        }
```

Comment:
As far as I can see, the only thing that we do here is to set the channel, and to enable the AP if it wasn't enabled yet. I think the disable, if it is needed, should be handled by bwl itself instead of by ap_manager, because a good implementation should be able to change the channel without disabling the AP (and it should also be able to tell clients to roam to the new channel).

And of course, I think all of this shouldn't be needed because we shouldn't manage the backhaul from prplMesh, but that's a different issue.